src/viam/sdk/components/private/board_server.cpp

```diff
@@ -174,6 +177,51 @@ ::grpc::Status BoardServer::GetDigitalInterruptValue(
     return ::grpc::Status();
 }
 
+::grpc::Status BoardServer::StreamTicks(
+    ::grpc::ServerContext* context,
+    const ::viam::component::board::v1::StreamTicksRequest* request,
+    ::grpc::ServerWriter<::viam::component::board::v1::StreamTicksResponse>* writer) {
+    if (!request) {
+        return ::grpc::Status(::grpc::StatusCode::INVALID_ARGUMENT,
+                              "Called [Board::StreamTicks] without a request");
+    };
+
+    const std::shared_ptr<Resource> rb = resource_manager()->resource(request->name());
+    if (!rb) {
+        return grpc::Status(grpc::UNKNOWN, "resource not found: " + request->name());
+    }
+
+    const std::shared_ptr<Board> board = std::dynamic_pointer_cast<Board>(rb);
+
+    AttributeMap extra;
+    if (request->has_extra()) {
+        extra = struct_to_map(request->extra());
+    }
+
+    const std::shared_ptr<std::queue<Board::tick>> ticks;
+    const std::vector<std::string> digital_interrupt_names(request->pin_names().begin(),
+                                                           request->pin_names().end());
+
+    board->stream_ticks(digital_interrupt_names, ticks, extra);
+
+    ::viam::component::board::v1::StreamTicksResponse response;
+    while (true) {
```

Comment:
Could this just be `while (!ticks->empty())`?

Reply:
No because this should still run even if the tick queue has nothing in it. If the queue is empty this should still be running as the server should also be listening for new ticks to come into the queue.